docs/architecture/adr-071-proposer-based-timestamps.md

```diff
@@ -336,73 +220,76 @@ The timeout in `enterPropose` will then be set as the maximum of `waitingTime` a
 
-The logic to validate that the block timestamp is greater than the previous block’s timestamp also works for proposer-based timestamps and will not change.
+#### Block timestamp validation logic common to both blocks that previously received a polka and those that did not
+
+Some block timestamp validation logic will apply to both reproposed blocks and newly proposed blocks.
+
+The current logic to validate that the block timestamp is greater than the previous block’s timestamp also works for proposer-based timestamps and will not change.
 
 `BFTTime` validation is no longer applicable and will be removed.
-Validators will no longer check that the block timestamp is a weighted median of `LastCommit` timestamps.
-This will mean removing the call to [MedianTime in the validateBlock function](https://github.com/tendermint/tendermint/blob/4db71da68e82d5cb732b235eeb2fd69d62114b45/state/validation.go#L117).
+This means that validators will no longer check that the block timestamp is a weighted median of `LastCommit` timestamps.
+Specifically, we will remove the call to [MedianTime in the validateBlock function](https://github.com/tendermint/tendermint/blob/4db71da68e82d5cb732b235eeb2fd69d62114b45/state/validation.go#L117).
 The `MedianTime` function can be completely removed. 
-The `LastCommit` timestamps may also be removed. 
 
-The `signedBytes` validation logic in `VerifyCommit` will be slightly altered.
-The `CommitSig`s in the block’s `LastCommit` will no longer each contain a timestamp.
-The validation logic will instead include the `LastCommit.Timestamp` in the hash of fields for generating the `signedBytes`.
-The cryptographic signatures included in the `CommitSig`s will then be checked against this `signedBytes` hash to authenticate the timestamp.
-Specifically, the `VerifyCommit` function will be updated to use this new timestamp.  
+Finally, since `CommitSig`s will no longer contain a timestamp, the validator authenticating a commit will no longer include the `CommitSig` timestamp in the hash of fields it builds to check against the cryptographic signature.
+
+#### Block timestamp validation when a has not received a polka
+
+The [POLRound](https://github.com/tendermint/tendermint/blob/68ca65f5d79905abd55ea999536b1a3685f9f19d/types/proposal.go#L29) in the `Proposal` message indicates which round the block originally received a polka.
+A negative value in the `POLRound` field indicates that the proposer has not seen a polka for this block.
+In this case, the validation logic will check for `timely`.
+
+The validation logic will be updated to check that the block timestamp is within `PRECISION` of the current Unix time known to the validator.
+If the timestamp is not within `PRECISION` of the current Unix time known to the validator, the proposed block will not be considered valid.
+
+If no valid proposal is received by the proposal timeout, the validator will prevote nil.
+This is identical to the current logic.
```

Comment:
This is incorrect. 

A message `m` is considered `timely` if received at `now_p < m.time + MSGDELAY + PRECISION`. So, a message `m` can be received before the `propose_timeout` event and still be `untimely`, in particular if the propose assigns a `m.time` that is in the past (lower than its current local time).

Reply:
Ah, this line was just referring to the the timeout logic. If a validator times out without receiving a proposal, it will prevote nil. This is true in both the current algorithm and the updated algorithm.

 I removed the line describing this as 'identical to the current logic' because it is ambiguous as to what it is referring to, although the intention was to just refer to prevoting nil in the case of not receiving a proposal by the timeout.